File: Medium/19. Remove Nth Node From End of List/Solution.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode aux = new ListNode(-1, head);
        ListNode front = aux;
        ListNode rear = aux;

        for (int i = 0; i <= n; i++)
            front = front.next;

        while (front != null) {
            front = front.next;
            rear = rear.next;
        }

        rear.next = rear.next.next;

        return aux.next;
    }
}

File: Medium/19. Remove Nth Node From End of List/solution.js
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function (head, n) {
  let aux = new ListNode(null, head);
  let front = aux, rear = aux;

  for (let i = 0; i <= n; i++)
    front = front.next;

  while (front) {
    front = front.next;
    rear = rear.next;
  }

  rear.next = rear.next.next;

  return aux.next;
};
